src/ILCompiler.DependencyAnalysisFramework/src/DependencyNodeCore.cs

```diff
@@ -138,5 +138,8 @@ protected virtual void OnMarked(DependencyContextType context)
         {
             // Do nothing by default
         }
+
+        // Force all non-abstract nodes to provide a name
+        protected internal abstract string GetName(DependencyContextType context);
```

Comment:
Can we just make this public instead of protected internal? Protected internal is rather odd, and we have no particular reason not to make this public.